openflow13/multipart.go

```diff
@@ -57,28 +64,45 @@ func (s *MultipartRequest) UnmarshalBinary(data []byte) error {
 	n += 2
 	n += 4 // for padding
 
-	var req util.Message
-	switch s.Type {
-	case MultipartType_Aggregate:
-		req = s.Body.(*AggregateStatsRequest)
-		err = req.UnmarshalBinary(data[n:])
-	case MultipartType_Desc:
-		break
-	case MultipartType_Flow:
-		req = s.Body.(*FlowStatsRequest)
-		err = req.UnmarshalBinary(data[n:])
-	case MultipartType_Port:
-		req = s.Body.(*PortStatsRequest)
-		err = req.UnmarshalBinary(data[n:])
-	case MultipartType_Table:
-		break
-	case MultipartType_Queue:
-		req = s.Body.(*QueueStatsRequest)
-		err = req.UnmarshalBinary(data[n:])
-	case MultipartType_Experimenter:
-		break
+	if n < s.Header.Length {
+		var req util.Message
+		switch s.Type {
+		case MultipartType_Aggregate:
+			req = new(AggregateStatsRequest)
+			err = req.UnmarshalBinary(data[n:])
+		case MultipartType_Desc:
+			break
+		case MultipartType_Flow:
+			req = new(FlowStatsRequest)
+			err = req.UnmarshalBinary(data[n:])
+		case MultipartType_Port:
+			req = new(PortStatsRequest)
+			err = req.UnmarshalBinary(data[n:])
+		case MultipartType_Table:
+			break
+		case MultipartType_Queue:
+			req = new(QueueStatsRequest)
+			err = req.UnmarshalBinary(data[n:])
+		case MultipartType_Experimenter:
+			break
+		case MultipartType_TableFeatures:
+			req = new(OFPTableFeatures)
+		}
+		if err != nil {
+			return err
+		}
+		if req != nil {
+			err = req.UnmarshalBinary(data[n:])
+			if err != nil {
+				return err
+			}
+			n += req.Len()
+			s.Body = req
+		} else {
+			return fmt.Errorf("unsupported MultipartRequest type: %d", s.Type)
+		}
```

Comment:
just a style nit...
I think we typically try to handle the error / early-return case first to reduce indentation overall
```go
if req == nil {
    return fmt.Errorf("unsupported MultipartRequest type: %d", s.Type)
}
if err := req.UnmarshalBinary(data[n:]); err != nil {
    return err
}
n += req.Len()
s.Body = req
```

Reply:
I wasn't sure, but after reading the OpenFLow spec, that works for me